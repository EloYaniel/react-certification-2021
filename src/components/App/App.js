import React from 'react';
import { createGlobalStyle } from 'styled-components';
import { Home } from '../Home';
import { Header } from '../Header';

const GlobalStyle = createGlobalStyle`
  * {
    margin: 0px;
    padding: 0px;
  }
  html {
    box-sizing: border-box;
    font-size: 62.5%;
  }
  body {
    background-color: #eee;
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
  }

  @media only screen and (max-width: 60em) {
    body {
      padding: 3rem;
    }
  }
`;

export const App = () => (
  <>
    <GlobalStyle />
    <Header />
    <Home />
  </>
);
