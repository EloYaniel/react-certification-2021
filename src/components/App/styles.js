import styled, { createGlobalStyle } from 'styled-components';

export const darkTheme = {
  textColor: '#fff',
  cardColor: '#424242',
  cardTitleColor: '#fff',
  cardDescriptionColor: '#fff',
  bodyBackgroundColor: '#303030',
  primaryColor: '',
  backgroundColor: '#000000',
};

export const lightTheme = {
  textColor: '#464e51',
  cardColor: '#fff',
  cardTitleColor: '#0d0d0d',
  cardDescriptionColor: '#3d3d3d',
  bodyBackgroundColor: '#eee',
  primaryColor: '',
  backgroundColor: '#1c5476',
};

export const GlobalStyle = createGlobalStyle`
  * {
    margin: 0px;
    padding: 0px;
  }

  html {
    box-sizing: border-box;
  }

  body {
    background-color: ${(props) => props.theme.bodyBackgroundColor};
    justify-content: center;
    align-items: center;
  }
`;

export const MainContainer = styled.div`
  margin: 50px;
`;
