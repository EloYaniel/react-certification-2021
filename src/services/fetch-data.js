export const fetchData = async (url) => {
  const response = await fetch(url, {
    headers: {
      'Content-Type': 'application/json',
    },
  });

  return response.json();
};
