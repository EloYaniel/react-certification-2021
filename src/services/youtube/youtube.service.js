import * as mockData from '../../utils/youtube-videos-mock.json';

export const getYoutubeMainInfoList = () => {
  const mainInfo = mockData.items.map((item) => ({
    id: item.etag,
    title: item.snippet.title,
    description: item.snippet.description,
    imageURL: item.snippet.thumbnails.default.url,
  }));

  return Promise.resolve(mainInfo);
};
