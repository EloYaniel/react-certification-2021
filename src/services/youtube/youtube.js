import { fetchData } from '../fetch-data';

export const getYoutubeMainInfoList = async (search) => {
  const url = `https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=18&q=${encodeURI(
    search || 'Wizeline'
  )}&type=video&key=AIzaSyBYzWLlzPQfofKh32frvUh6oKJgB98ghis`;
  const response = await fetchData(url);
  const mainInfo = response.items.map((item) => ({
    id: item.id.videoId,
    title: item.snippet.title,
    description: item.snippet.description,
    imageURL: item.snippet.thumbnails.high.url,
  }));

  return Promise.resolve(mainInfo);
};

export const getRelatedVideos = async (relatedToVideoId) => {
  const url = `https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=18&relatedToVideoId=${relatedToVideoId}&type=video&key=AIzaSyBYzWLlzPQfofKh32frvUh6oKJgB98ghis`;
  const response = await fetchData(url);

  const mainInfo = response.items.map((item) => ({
    id: item.id.videoId,
    title: item.snippet.title,
    description: item.snippet.description,
    imageURL: item.snippet.thumbnails.default.url,
  }));

  return Promise.resolve(mainInfo);
};

export const getFavoriteVideos = async (videoIDs) => {
  let url =
    'https://youtube.googleapis.com/youtube/v3/videos?part=snippet&key=AIzaSyBYzWLlzPQfofKh32frvUh6oKJgB98ghis';
  videoIDs.forEach((id) => {
    url = `${url}&id=${id}`;
  });
  const response = await fetchData(url);

  const mainInfo = response.items.map((item) => ({
    id: item.id,
    title: item.snippet.title,
    description: item.snippet.description,
    imageURL: item.snippet.thumbnails.default.url,
  }));

  return Promise.resolve(mainInfo);
};

export const getVideoInfo = async (videoID) => {
  const url = `https://youtube.googleapis.com/youtube/v3/videos?part=snippet&key=AIzaSyBYzWLlzPQfofKh32frvUh6oKJgB98ghis&id=${videoID}`;
  const response = await fetchData(url);

  const mainInfo = response.items.map((item) => ({
    id: item.id,
    title: item.snippet.title,
    description: item.snippet.description,
    imageURL: item.snippet.thumbnails.default.url,
  }));

  return Promise.resolve(mainInfo);
};
