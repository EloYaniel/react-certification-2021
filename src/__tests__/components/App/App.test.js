import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { App } from '../../../components/App';
import { getYoutubeMainInfoList } from '../../../services/youtube';

jest.mock('../../../services/youtube');

describe('App Component tests suit', () => {
  it('Should match snapshot', async () => {
    getYoutubeMainInfoList.mockImplementation(() =>
      Promise.resolve([
        {
          id: '123',
          title: 'Wizeline Co.',
          description: 'Awesome Company',
          imageURL: 'http://wizeline.com',
        },
      ])
    );

    const element = render(<App />, { wrapper: MemoryRouter });

    await waitFor(() => {
      screen.getByText('Wizeline Co.');
      expect(element).toMatchSnapshot();
    });

    getYoutubeMainInfoList.mockRestore();
  });
});
