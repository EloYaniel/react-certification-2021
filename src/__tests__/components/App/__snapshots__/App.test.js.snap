// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component tests suit Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-idOhPF enrezt"
      >
        <div
          class="sc-hHftDr fjyVqu"
        >
          <div
            class="sc-kEjbxe dHxCwD"
          >
            <div
              class="sc-iqHYGH cOCNrb"
            />
            <div
              class="sc-iqHYGH cOCNrb"
            />
            <div
              class="sc-iqHYGH cOCNrb"
            />
          </div>
          <input
            class="sc-kstrdz cZxlvy"
            placeholder="Wizeline"
          />
        </div>
        <div
          class="sc-dIUggk kOwbcQ"
        >
          <label
            class="sc-hBEYos iufiwT"
          >
            <input
              class="sc-fodVxV leZTiF"
              type="checkbox"
              value="false"
            />
            <div
              class="sc-fFubgz eQvTUS"
            />
            <span
              class="sc-bkzZxe drHitx"
            >
              Dark mode
            </span>
          </label>
          <a
            class="sc-pFZIQ clLSRs"
            href="/login"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </a>
        </div>
      </header>
      <div
        class="sc-bYEvPH EmZQK"
      >
        <h1
          class="sc-fubCfw gyQTdR"
        >
          Welcome to my challenges!!!
        </h1>
        <div
          class="sc-iBPRYJ cWUbzV"
        >
          <div
            class="sc-bdfBwQ bttSXB"
          >
            <div
              class="sc-gsTCUz"
            >
              <img
                alt="Wizeline Co."
                class="sc-gKsewC cFFDOc"
                src="http://wizeline.com"
              />
              <div
                class="sc-dlfnbm ilKvLc"
              >
                <h1
                  class="sc-hKgILt dVnqlC"
                >
                  Wizeline Co.
                </h1>
                <p
                  class="sc-eCssSg byBHyx"
                >
                  Awesome Company
                </p>
              </div>
              <a
                class="sc-jSgupP clLjnz"
                href="/detail/123"
              >
                View More
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-idOhPF enrezt"
    >
      <div
        class="sc-hHftDr fjyVqu"
      >
        <div
          class="sc-kEjbxe dHxCwD"
        >
          <div
            class="sc-iqHYGH cOCNrb"
          />
          <div
            class="sc-iqHYGH cOCNrb"
          />
          <div
            class="sc-iqHYGH cOCNrb"
          />
        </div>
        <input
          class="sc-kstrdz cZxlvy"
          placeholder="Wizeline"
        />
      </div>
      <div
        class="sc-dIUggk kOwbcQ"
      >
        <label
          class="sc-hBEYos iufiwT"
        >
          <input
            class="sc-fodVxV leZTiF"
            type="checkbox"
            value="false"
          />
          <div
            class="sc-fFubgz eQvTUS"
          />
          <span
            class="sc-bkzZxe drHitx"
          >
            Dark mode
          </span>
        </label>
        <a
          class="sc-pFZIQ clLSRs"
          href="/login"
        >
          <svg
            aria-hidden="true"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </a>
      </div>
    </header>
    <div
      class="sc-bYEvPH EmZQK"
    >
      <h1
        class="sc-fubCfw gyQTdR"
      >
        Welcome to my challenges!!!
      </h1>
      <div
        class="sc-iBPRYJ cWUbzV"
      >
        <div
          class="sc-bdfBwQ bttSXB"
        >
          <div
            class="sc-gsTCUz"
          >
            <img
              alt="Wizeline Co."
              class="sc-gKsewC cFFDOc"
              src="http://wizeline.com"
            />
            <div
              class="sc-dlfnbm ilKvLc"
            >
              <h1
                class="sc-hKgILt dVnqlC"
              >
                Wizeline Co.
              </h1>
              <p
                class="sc-eCssSg byBHyx"
              >
                Awesome Company
              </p>
            </div>
            <a
              class="sc-jSgupP clLjnz"
              href="/detail/123"
            >
              View More
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
